tests/receivers/jmx/cassandra/jmx_cassandra_discovery_test.go

```diff
@@ -66,8 +69,13 @@ func jmxCassandraAutoDiscoveryHelper(t *testing.T, ctx context.Context, configFi
 	factory := otlpreceiver.NewFactory()
 	port := 16745
 	c := factory.CreateDefaultConfig().(*otlpreceiver.Config)
-	c.GRPC.NetAddr.Endpoint = fmt.Sprintf("localhost:%d", port)
-	endpoint := c.GRPC.NetAddr.Endpoint
+	endpoint := fmt.Sprintf("localhost:%d", port)
+	c.GRPC = configoptional.Some(configgrpc.ServerConfig{
+		NetAddr: confignet.AddrConfig{
+			Endpoint:  endpoint,
```

Comment:
```suggestion
	c.GRPC = configoptional.Some(configgrpc.ServerConfig{
		NetAddr: confignet.AddrConfig{
			Endpoint:  fmt.Sprintf("localhost:%d", port),
```
Nit, feel free to drop: Other tests just set endpoint directly, we can do the same here for consistency.

Reply:
It is passed down as an environment variable, that's why it is saved in the local `endpoint`.